### PR TITLE
Update mappings to repair failing alert creation and matching dependancies

### PR DESCRIPTION
**Pull request by Killklli**
I'm really not a fan of what I've done here, but this likely should mitigate the issues with dependencyalert not being initialized correctly. 
This is blocking https://github.com/advanced-security/policy-as-code/pull/121

### CODE REVIEW

File: src/ghastoolkit/supplychain/dependencies.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def getPurl(self, version: bool = True) -> str:
         if self.manager:
             result += f"{self.manager.lower()}/"
         if self.namespace:
-            result += f"{self.namespace}/"
+            result += f"{self.namespace}:"
```
Comment (GeekMasher) [inline src/ghastoolkit/supplychain/dependencies.py:54]: The `/` here is used to separate the namespace and name of the packages.

https://github.com/package-url/purl-spec?tab=readme-ov-file#purl

Is there a reason to change it?

**Review by GeekMasher — APPROVED** [review]
Apart from one comment, rest looks great

File: src/ghastoolkit/supplychain/dependencies.py (hunk 1 above)
Comment (Killklli) [reply src/ghastoolkit/supplychain/dependencies.py:54]: Yes, the formatting coming back from the perl lookup in policy as code returns it with : not a / so for me to be able to do proper comparisons there I made that change

Comment (Killklli) [reply src/ghastoolkit/supplychain/dependencies.py]: So if you look at, https://github.com/GeekMasher/ghastoolkit/pull/232/files#diff-5c97a427ff057c148668cb63e5aea09904fddb4c6494c8a2a1442d5f550266e7R97 that response is coming from the dependabot, the name of the package itself is being provided as x:y vs doing x/y so I had swapped to it for that as thats an upstream result, would you prefer I just do a find and replace?

Comment (GeekMasher) [reply src/ghastoolkit/supplychain/dependencies.py:54]: Thanks for tracking this down @Killklli, very strange its coming from Dependabot / DepGraph. Thank you for resolving this

Comment (GeekMasher) [reply src/ghastoolkit/supplychain/dependencies.py]: For now I have removed this small change as the rest of the code looks good. Lets talk offline to discuss this change more

**Pull request merged**